backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/MetadataUtils.java

```diff
@@ -249,16 +254,43 @@ private static void createRowLevelPermissions(DSLContext jooq, org.jooq.Table ta
 
   protected static void saveSchemaMetadata(DSLContext sql, SchemaMetadata schema) {
     try {
-      sql.insertInto(SCHEMA_METADATA).columns(TABLE_SCHEMA).values(schema.getName()).execute();
+      String description = Objects.isNull(schema.getDescription()) ? "" : schema.getDescription();
+      sql.insertInto(SCHEMA_METADATA)
+          .columns(TABLE_SCHEMA, SCHEMA_DESCRIPTION)
+          .values(schema.getName(), description)
+          .execute();
     } catch (Exception e) {
       throw new MolgenisException("save of schema metadata failed", e);
     }
   }
 
+  protected static void updateSchemaMetadata(DSLContext sql, SchemaMetadata schema) {
+    try {
+      sql.update(SCHEMA_METADATA)
+          .set(SCHEMA_DESCRIPTION, schema.getDescription())
+          .where(TABLE_SCHEMA.eq(schema.getName()))
+          .execute();
+    } catch (Exception e) {
+      throw new MolgenisException("update of schema metadata failed", e);
+    }
+  }
+
   protected static Collection<String> loadSchemaNames(SqlDatabase db) {
     return db.getJooq().selectFrom(SCHEMA_METADATA).fetch().getValues(TABLE_SCHEMA, String.class);
   }
 
+  protected static Collection<SchemaInfo> loadSchemaInfos(SqlDatabase db) {
```

Comment:
I would propose we merge loadSchemaNames and loadSchemaMetadata (that you call loadSchemaInfos).

And then make SqlSchemaMetadata 'lazy' so it only loads its column/setting metadata when used.

Then we can get rid of the silly double cahcing of 'schema names' list and 'schema metadata' map in Database

Reply:
see points above